### PR TITLE
Extract custom clearing price logic

### PR DESCRIPTION
**Pull request by fleupold**
# Description
For the domain based settlement encoding we need to encode both the uniform clearing prices of the solution as well as the custom (fee adjusted) clearing prices of each trade. This PR create a utility function on `Trade` that allows this logic to be shared.

After rebasing, I noticed @Mateo-mro already introduced a similar function (`calculate_custom_price`). However, this one wasn't reusing the existing math to calculate net buy and sell amounts and had a slightly longer name, so this is now mainly a refactoring/renaming of this method.

# Changes
- [x] Move `CustomClearingPrice` type to up since it won't be `scoring` specific
- [x] Reuse buy/sell amount calculation to derive the clearing price
- [x] Rename use of `calculate_custom_price`

## How to test
CI

## Related Issues

Part of #2215

### CODE REVIEW

**Comment by sunce86** [comment]
I think we did basically the same thing: https://github.com/cowprotocol/services/pull/2658 I am fine with merging both. I can rebase on top of your PR since my PR also refactors the `scoring` file.

**Comment by fleupold** [comment]
In that case, let's prefer your work (superseeded by 2658).

**Comment by fleupold** [comment]
Actually moving the CustomClearingPrice type is still relevant, so maybe we still want to merge this one

File: crates/driver/src/domain/competition/solution/trade.rs [hunk 1]
```diff
@@ -164,20 +131,30 @@ impl Fulfillment {
...
                 .checked_mul(prices.sell)
                 .ok_or(Math::Overflow)?
-                .checked_div(prices.buy)
+                .checked_ceil_div(&prices.buy)
                 .ok_or(Math::DivisionByZero)?,
         };
         Ok(eth::TokenAmount(amount))
     }
 
+    pub fn custom_prices(
```
Comment (sunce86) [inline crates/driver/src/domain/competition/solution/trade.rs:148]: nit: maybe add comment on pub function

File: crates/driver/src/domain/competition/solution/trade.rs [hunk 2]
```diff
@@ -164,20 +131,30 @@ impl Fulfillment {
...
+    pub fn buy_amount(&self, prices: &ClearingPrices) -> Result<eth::TokenAmount, error::Math> {
         let amount = match self.order.side {
             order::Side::Buy => self.executed.0,
             order::Side::Sell => self
                 .executed
                 .0
                 .checked_mul(prices.sell)
                 .ok_or(Math::Overflow)?
-                .checked_div(prices.buy)
+                .checked_ceil_div(&prices.buy)
```
Comment (sunce86) [inline crates/driver/src/domain/competition/solution/trade.rs:142]: 👍

**Review by sunce86 — APPROVED** [review]
LG

**Review by MartinquaXD — APPROVED** [review]

**Review by squadgazzz — APPROVED** [review]

**Pull request merged**